beacon-chain/chaintest/backend/helpers.go

```diff
@@ -0,0 +1,122 @@
+package backend
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/gogo/protobuf/proto"
+	b "github.com/prysmaticlabs/prysm/beacon-chain/core/blocks"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// Generates a simulated beacon block to use
+// in the next state transition given the current state,
+// the previous beacon block, and previous beacon block root.
+func generateSimulatedBlock(
+	beaconState *pb.BeaconState,
+	prevBlockRoot [32]byte,
+	randaoReveal [32]byte,
+) (*pb.BeaconBlock, [32]byte, error) {
+	encodedState, err := proto.Marshal(beaconState)
+	if err != nil {
+		return nil, [32]byte{}, fmt.Errorf("could not marshal beacon state: %v", err)
+	}
+	stateRoot := hashutil.Hash(encodedState)
+	block := &pb.BeaconBlock{
+		Slot:               beaconState.Slot + 1,
+		RandaoRevealHash32: randaoReveal[:],
+		ParentRootHash32:   prevBlockRoot[:],
+		StateRootHash32:    stateRoot[:],
+		Body: &pb.BeaconBlockBody{
+			ProposerSlashings: []*pb.ProposerSlashing{},
```

Comment:
shouldn't these block operations (e.g. slashing, attestation..etc) be part of the input parameter?

Reply:
Will be done through multiple PRs in which they will be indeed added as parameters